docs/content/docs/api-reference/workflow-ai/durable-agent.mdx

```diff
@@ -177,6 +184,52 @@ async function multiToolAgentWorkflow(userQuery: string) {
 }
 ```
 
+### Multi-turn Conversation
+
+```typescript
+import { DurableAgent } from '@workflow/ai/agent';
+import { z } from 'zod';
+
+async function searchProducts({ query }: { query: string }) {
+  "use step";
+  // Search product database
+  return `Found 3 products matching "${query}"`;
+}
+
+async function multiTurnAgentWorkflow() {
+  'use workflow';
+
+  const agent = new DurableAgent({
+    model: 'anthropic/claude-haiku-4.5',
+    tools: {
+      searchProducts: {
+        description: 'Search for products',
+        inputSchema: z.object({ query: z.string() }),
+        execute: searchProducts,
+      },
+    },
+  });
+
```

Comment:
we should wire the writable through  here too and explain that the messages array is returned as a convenience, in addition to streaming to the writable, so that the user can do stateful stuff.

We can even get rid of `messages` in the previous example above and only use it in this example (to keep the previous example lighter)